diffkemp/diffkemp.py

```diff
@@ -18,222 +22,266 @@
-        source_finder_cls = SingleLlvmFinder
-        source_finder_path = args.single_llvm_file
+def build(args):
+    # Generate wrapper for C/C++ compiler
+    cc_wrapper = get_cc_wrapper_path(args.no_native_cc_wrapper)
+
+    # Create temp directory and environment
+    tmpdir = mkdtemp()
+    db_filename = os.path.join(tmpdir, "diffkemp-wdb")
+    environment = {
+        wrapper_env_vars["db_filename"]: db_filename,
+        wrapper_env_vars["clang"]: args.clang,
+        wrapper_env_vars["clang_append"]: ",".join(args.clang_append)
+        if args.clang_append is not None
+        else "",
+        wrapper_env_vars["clang_drop"]: ",".join(args.clang_drop)
+        if args.clang_drop is not None
+        else "",
+        wrapper_env_vars["llvm_link"]: args.llvm_link,
+    }
+    environment.update(os.environ)
+
+    # Determine make args
+    make_cc_setting = 'CC="{}"'.format(cc_wrapper)
+    make_args = [args.build_program, "-C", args.source_dir, make_cc_setting]
+    if args.build_file is not None:
+        make_args.extend(["-c", args.build_file])
+    make_target_args = make_args[:]
+    if args.target is not None:
+        make_target_args.extend(args.target)
+
+    # Clean the project
+    config_log_filename = os.path.join(args.source_dir, "config.log")
+    if os.path.exists(config_log_filename):
+        # Backup config.log
+        os.rename(config_log_filename, config_log_filename + ".bak")
+    try:
+        check_call(make_args + ["clean"], env=environment)
+    except CalledProcessError:
+        pass
+    if os.path.exists(config_log_filename + ".bak"):
+        # Restore config.log
+        os.rename(config_log_filename + ".bak", config_log_filename)
+
+    # If config.log is present, reconfigure using wrapper
+    # Note: this is done to support building with nested configure
+    if args.reconfigure and os.path.exists(config_log_filename):
+        with open(config_log_filename, "r") as config_log:
+            # Try to get line with configure command from config.log
+            # This line is identified by being the first line containing $
+            configure_cmd = None
+            for line in config_log.readlines():
+                if "$" in line:
+                    configure_cmd = line.strip()
+                    break
+            if configure_cmd and make_cc_setting not in configure_cmd:
+                # Remove beginning of line containing spaces and $
+                configure_cmd = configure_cmd.split("$ ", 1)[1]
+                configure_cmd += " " + make_cc_setting
+                # Delete all config.cache files
+                for root, _, files in os.walk(args.source_dir):
+                    for filename in files:
+                        if filename != "config.cache":
+                            continue
+                        try:
+                            os.remove(os.path.join(root, filename))
+                        except PermissionError:
+                            pass
```

Comment:
Maybe more clear like this?
```suggestion
                    if "config.cache" in files:
                        try:
                            os.remove(os.path.join(root, "config.cache"))
                        except PermissionError:
                            pass
```

Reply:
Actually I should have simply iterated on directories and done something like
```python
try:
    os.remove(os.path.join(dirname, "config.cache")
except (FileNotFoundError, PermissionError):
    pass
```
I got confused by somehow automatically trying to replicate exactly what the shell one-liner did, which is less efficient on modern filesystems, since `os.remove` just looks up the file in a B-tree or a similar structure once, no need to go through all files in all directories in the tree.